databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,471 @@
+        and will be executed.
+        All nodes should have a unique key, and this method will try to create unique index on
+        the LABEL when it sees first time within a job scope.
+        Example of Cypher query executed by this method:
+        MERGE (col_test_id1:Column {key: 'presto://gold.test_schema1/test_table1/test_id1'})
+        ON CREATE SET col_test_id1.name = 'test_id1',
+                      col_test_id1.order_pos = 2,
+                      col_test_id1.type = 'bigint'
+        ON MATCH SET col_test_id1.name = 'test_id1',
+                     col_test_id1.order_pos = 2,
+                     col_test_id1.type = 'bigint'
+
+        :param node_file:
+        :return:
+        """
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            all_node_records = pandas.read_csv(node_csv, na_filter=False).to_dict(orient="records")
+            if len(all_node_records) > 0:
+                stmt = self.create_node_merge_statement(node_record=all_node_records[0])
+
+            self._write_transactions(all_node_records, stmt)
+
+    def create_node_merge_statement(self, node_record: dict) -> str:
+        """
+        Creates node merge statement
+        :param node_record:
+        :return:
+        """
+        template = Template("""
+            UNWIND $batch AS row
+            MERGE (node:{{ LABEL }} {key: row.KEY})
+            ON CREATE SET {{ PROP_BODY_CREATE }}
+            {% if update %} ON MATCH SET {{ PROP_BODY_UPDATE }} {% endif %}
+        """)
+
+        prop_body_create = self._create_props_body(node_record, NODE_REQUIRED_KEYS, 'node')
+
+        prop_body_update = prop_body_create
+        if self._preserve_adhoc_ui_data:
+            prop_body_update = self._create_props_body(node_record, NODE_REQUIRED_KEYS, 'node', True)
+
+        return template.render(LABEL=node_record[NODE_LABEL],
+                               PROP_BODY_CREATE=prop_body_create,
+                               PROP_BODY_UPDATE=prop_body_update,
+                               update=(not node_record[NODE_LABEL] in self._create_only_nodes))
+
+    def _publish_relation(self, relation_file: str) -> None:
+        """
+        Creates relation between two nodes.
+        (In Amundsen, all relation is bi-directional)
+
+        Example of Cypher query executed by this method:
+        MATCH (n1:Table {key: 'presto://gold.test_schema1/test_table1'}),
+              (n2:Column {key: 'presto://gold.test_schema1/test_table1/test_col1'})
+        MERGE (n1)-[r1:COLUMN]->(n2)-[r2:BELONG_TO_TABLE]->(n1)
+        RETURN n1.key, n2.key
+
+        :param relation_file:
+        :return:
+        """
+
+        with open(relation_file, 'r', encoding='utf8') as relation_csv:
+            all_rel_records = pandas.read_csv(relation_csv, na_filter=False).to_dict(orient="records")
+            if len(all_rel_records) > 0:
+                stmt = self.create_relationship_merge_statement(rel_record=all_rel_records[0])
```

Comment:
it's a bit confusing that this takes a full record. suggestion:

```
csv_dataframe = pandas.read_csv(relation_csv, na_filter=False)
merge_statement = self.create_relationship_merge_statement(csv_dataframe.columns.tolist())
self._write_transactions(merge_statement, csv_dataframe.to_dict(orient="records"))
```
ref: https://stackoverflow.com/questions/24962908/how-can-i-read-only-the-header-column-of-a-csv-file-using-python